src/storage-client/src/controller.rs

```diff
@@ -622,11 +624,19 @@ pub struct StorageControllerState<
     stashed_response: Option<StorageResponse<T>>,
     /// Storage hosts that should be deprovisioned during the next call to `StorageController::process`.
     pending_host_deprovisions: BTreeSet<GlobalId>,
+    /// Commands that should be send to storage instances during the next call
+    /// to `StorageController::process`.
+    pending_compaction_commands: Vec<(GlobalId, Antichain<T>)>,
 
     /// Interface for managed collections
     pub(super) collection_manager: collection_mgmt::CollectionManager,
     /// Tracks which collection is responsible for which [`IntrospectionType`].
     pub(super) introspection_ids: HashMap<IntrospectionType, GlobalId>,
+    /// Tokens for tasks that drive updating introspection collections. Dropping
+    /// this will make sure that any tasks (or other resources) will stop when
+    /// needed.
+    // TODO(aljoscha): Should these live somewhere else?
+    introspection_tokens: HashMap<GlobalId, Arc<dyn Any + Send + Sync>>,
```

Comment:
i'd use `Box`, not `Arc`; this doesn't need to be `: Clone`

Also, I agree that this a weird spot for this, but I can't think of any better place

Reply:
Also, this should just be a `Vec` of the values, its never cleaned by-key, so doesn't need to be in a hashmap!

Reply:
I made it a `HashMap` to convey the intention that we might want to clean these up by `GlobalId`, but I'm changing to `Box` 👌

Reply:
sounds good!